src/charm.py

```diff
@@ -78,6 +111,181 @@ def configure_magma_access_gateway(orc8r_domain: str, root_ca_path: str) -> None
+            logger.warning("Invalid IPv6 address and netmask for interface sgi")
+            return False
+        if ipv6_gateway and not self._is_valid_ipv6_gateway(ipv6_gateway):
+            logger.warning("Invalid IPv6 gateway for interface sgi")
+            return False
+        return True
+
+    def _is_valid_s1_interface_addressing_configuration(self) -> bool:
+        """Validates s1 interface configuration."""
+        ipv4_address = self.model.config.get("s1-ipv4-address")
+        ipv6_address = self.model.config.get("s1-ipv6-address")
+        if not ipv4_address and not ipv6_address:
+            return True
+        if ipv6_address and not ipv4_address:
+            logger.warning("Pure IPv6 configuration is not supported for interface s1")
+            return False
+        if ipv4_address and not self._is_valid_ipv4_address(ipv4_address):
+            logger.warning("Invalid IPv4 address and netmask for interface s1")
+            return False
+        if ipv6_address and not self._is_valid_ipv6_address(ipv6_address):
+            logger.warning("Invalid IPv6 address and netmask for interface s1")
+            return False
+        return True
+
+    @staticmethod
+    def _is_valid_ipv4_address(ipv4_address: str) -> bool:
+        """Validate an IPv4 address and netmask.
+
+        A valid string will have the form:
+        a.b.c.d/x
+        """
+        try:
+            ip = ipaddress.ip_network(ipv4_address, strict=False)
+            return isinstance(ip, ipaddress.IPv4Network) and ip.prefixlen != 32
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _is_valid_ipv4_gateway(ipv4_gateway: str) -> bool:
+        """Validate an IPv4 gateway.
+
+        A valid string will have the form:
+        a.b.c.d
+        """
+        try:
+            ip = ipaddress.ip_address(ipv4_gateway)
+            return isinstance(ip, ipaddress.IPv4Address)
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _is_valid_ipv6_address(ipv6_address: str) -> bool:
+        """Validate an IPv6 address and netmask.
+
+        A valid string will contain an IPv6 address followed by a
+        netmask, like this:
+        2001:0db8:85a3:0000:0000:8a2e:0370:7334/64
+        """
+        try:
+            ip = ipaddress.ip_network(ipv6_address, strict=False)
+            return isinstance(ip, ipaddress.IPv6Network) and ip.prefixlen != 128
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _is_valid_ipv6_gateway(ipv6_gateway: str) -> bool:
+        """Validate an IPv6 gateway.
+
+        A valid string will contain an IPv6 address, like this:
+        2001:0db8:85a3:0000:0000:8a2e:0370:7334
+        """
+        try:
+            ip = ipaddress.ip_address(ipv6_gateway)
+            return isinstance(ip, ipaddress.IPv6Address)
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _are_valid_dns(dns: str) -> bool:
```

Comment:
Is it intended that an empty list of DNS servers would be considered as valid

Reply:
No, good catch

Reply:
Fixed